inference-engine/thirdparty/clDNN/api/cldnn/runtime/engine.hpp

```diff
@@ -142,7 +142,7 @@ class engine {
     const device::ptr _device;
     engine_configuration _configuration;
 
-    std::atomic<uint64_t> memory_usage = {0};
+    std::map<allocation_type, std::atomic<uint64_t>> memory_usage_map;
     std::atomic<uint64_t> peak_memory_usage = {0};
```

Comment:
I think peak memory should be managed separately too

Reply:
done